etna/analysis/plotters.py

```diff
@@ -1312,12 +1322,61 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
     return holidays_df
 
 
+@_create_holidays_df.register
+def _create_holidays_df_dataframe(holidays: pd.DataFrame, index, as_is):
+    holidays_df = pd.DataFrame(index=index, columns=holidays["holiday"].unique(), data=0)
+
+    if as_is:
+        dt = holidays_df.index.intersection(holidays.index)
+        holidays_df.loc[dt, :] = holidays.loc[dt, :]
+        return holidays_df
+
+    for name in holidays["holiday"].unique():
+        ds = holidays[holidays["holiday"] == name]["ds"]
+        dt = [ds]
+        if "upper_window" in holidays.columns:
+            ds_upper_bound = ds + pd.to_timedelta(
```

Comment:
Will it work correctly if `ts` will have 15-minute frequency for example?

Try:
```python
pd.to_timedelta(10, unit="15T")
```

It fails.

We should add smth like these into tests.

Reply:
I fixed it and added test for this case